task17/index.js

```diff
@@ -0,0 +1,107 @@
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  try {
+    const value = await fetchImgData();
+    renderSlideImage(value);
+    renderSlideNumber(value);
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+};
+init();
+
+const renderSlideImage = (value) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImage.src = value[i].img;
+    slideImage.alt = value[i].alt;
+    value[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideImgList.appendChild(slideImgFragment);
+};
+
+const renderSlideNumber = (value) => {
+  const slideNumText = createElementWithClassName("p", "slide-number");
+  const slideLength = value.length;
+  slideNumText.id = "js-slideNumber";
+  currentImgIndex = value.findIndex((v) => v.display);
+  slideNumText.textContent = `${currentImgIndex + 1} / ${slideLength}`;
+  slideShowContainer.appendChild(slideNumText);
+};
+
+const slideBtn = document.querySelectorAll(".slide-btn");
+const nextBtn = document.getElementById("js-nextBtn");
+const prevBtn = document.getElementById("js-prevBtn");
+let currentImgIndex = 0;
+
+slideBtn.forEach((button) => {
+  prevBtn.disabled = currentImgIndex === 0;
+  button.addEventListener("click", (event) => {
+    const eventTarget = event.target.parentElement;
+    const slideImg = document.querySelectorAll(".slide-img-item");
+    document.querySelector(".is-active").classList.remove("is-active");
+    eventTarget.id === "js-nextBtn" ? ++currentImgIndex : --currentImgIndex;
+    slideImg[currentImgIndex].classList.add("is-active");
+    btnDisabled(slideImg);
+    renderActiveNumber(slideImg);
+  });
+});
+
+const btnDisabled = (value) => {
```

Comment:
IMO
It would be better to have a name that indicates that the `disabled` is to be switched.
```suggestion
const toggleDisabledOfButton = (value) => {
```

Reply:
Indeed.
The name has been corrected.
https://github.com/seeeen044/JavaScript_handson/commit/b40f5885a0c9b028a5a8c81238f1d7bd0448f741